src/compiler/transformers/jsx.ts

```diff
@@ -444,7 +446,7 @@ export function transformJsx(context: TransformationContext): (x: SourceFile | B
 
     function transformJsxSpreadAttributeToProps(node: JsxSpreadAttribute) {
         if (isObjectLiteralExpression(node.expression) && !hasProto(node.expression)) {
-            return node.expression.properties;
+            return sameMap(node.expression.properties, p => Debug.checkDefined(visitNode(p, visitor, isObjectLiteralElementLike)));

```

Comment:
This seems reasonable, but I can't help but notice that `sameMap` is never used in `**/transformers/**`, so I get a bad feeling that there's some other canonical method for doing this...

Reply:
Using this helper wasn't super intentional on my side. I just knew that there is such an util function and since this was returned as-is here before I decided to use it here (as they were returned as-is returning the same array shouldn't be the problem). 

I can change this to a regular `map` though - the outcome would be the same.